services/consensuscontext/validate_transactions_block_test.go

```diff
@@ -172,6 +199,36 @@ func TestConsensusContextValidateTransactionsBlock_ForForwardedToPreOrderErrors(
 	})
 }
 
+func TestConsensusContextValidateTransactionsBlock_ValidateBlockRefTime(t *testing.T) {
+	with.Context(func(ctx context.Context) {
+		cfg := config.ForConsensusContextTests(false)
+
+		t.Run("should NOT return error for new ref is newer than ref and not smaller than grace allows", func(t *testing.T) {
+			proposedRef := primitives.TimestampSeconds(time.Now().Unix())
+			prevRef := proposedRef - 500
+			validatorRef := proposedRef
+			err := validateProposeBlockReferenceTime(prevRef, proposedRef, validatorRef, cfg.ManagementConsensusGraceTimeout())
+			require.NoError(t, err)
+		})
+
+		t.Run("should return error for new ref is NOT newer than prev", func(t *testing.T) {
```

Comment:
should return error for new ref is less than prev. (equal is ok)

Reply:
not newer = equal or less